model/collection.py

```diff
@@ -744,9 +742,9 @@ def restrict_to_ready_deliverable_works(
         if not show_suppressed:
             query = query.filter(LicensePool.suppressed==False)
 
-        # Only find books with available licenses.
+        # Only find books with available licenses or books from self-hosted collections using MirrorUploader
         query = query.filter(
-                or_(LicensePool.licenses_owned > 0, LicensePool.open_access)
+            or_(LicensePool.licenses_owned > 0, LicensePool.open_access, LicensePool.self_hosted)
```

Comment:
In core/external_search.py:Filter#build there is equivalent code for putting this restriction into a query against the Elasticsearch index. Something similar happens in `universal_nested_filters. To get this to work, `open_access` is defined as a boolean field of the "licensepools" sub-document, and "self_hosted" should be added there as well. 

Basically, you should be add a self-hosted book to the data sets defined in `core/tests/test_external_search:TestExternalSearchWithWorks` and `TestFacetFilters`, and verify that the self-hosted book shows up in normal queries. I expect that such a book would not show up in queries now. 

Comprehensive documentation of our search index is [here](https://github.com/NYPL-Simplified/Simplified/wiki/Elasticsearch); I can also answer any questions you have. Once this branch lands I'll update the wiki page.

Reply:
I added an additional condition to `Work.to_search_document` using `LicensePool.self_hosted`:
```python
licensepools = select(
            [
                LicensePool.id.label('licensepool_id'),
                LicensePool.data_source_id.label('data_source_id'),
                LicensePool.collection_id.label('collection_id'),
                LicensePool.open_access.label('open_access'),
                LicensePool.suppressed,
                or_(
                    LicensePool.self_hosted,
                    LicensePool.licenses_available > 0,
                ).label('available'),
                (LicensePool.licenses_owned > 0).label('licensed'),
                work_quality_column,
                Edition.medium,
                func.extract(
                    "EPOCH",
                    LicensePool.availability_time,
                ).label('availability_time')
            ]
        ).where(
            and_(
                LicensePool.work_id==work_id_column,
                work_presentation_edition_id_column==Edition.id,
                or_(
                    LicensePool.open_access,
                    # LicensePool.self_hosted,
                    LicensePool.licenses_owned>0,
                ),
            )
        ).alias("licensepools_subquery")
```
However, I'm not completely sure whether I need to add `LicensePool.self_hosted` to **where** statement